From 8fbd158e31b97284af4ab691e19c0e3fb4f9f259 Mon Sep 17 00:00:00 2001
From: Priyanka Bose <priyanka.bose@intel.com>
Date: Fri, 15 May 2020 02:19:11 +0530
Subject: [PATCH] Disabled avx avx2 avx512 compiler flags by default. Library
 components can enable the flags based on performance impact

Change-Id: I41c4464cffd2b0e830d4ab1c5a25b378e2a25129
Tracked-On: https://jira.devtools.intel.com/browse/OAM-90793
Signed-off-by: Priyanka Bose <priyanka.bose@intel.com>
---
 cc/config/x86_64_device.go | 13 +++++++------
 cc/config/x86_device.go    | 20 ++++++++++----------
 2 files changed, 17 insertions(+), 16 deletions(-)

diff --git a/cc/config/x86_64_device.go b/cc/config/x86_64_device.go
index e1d418b..4888a76 100644
--- a/cc/config/x86_64_device.go
+++ b/cc/config/x86_64_device.go
@@ -48,15 +48,14 @@ var (
 		"ivybridge": []string{
 			"-march=core-avx-i",
 		},
-                "kabylake": []string{
-                       "-march=core-avx2",
-                },
 		"sandybridge": []string{
 			"-march=corei7",
 		},
 		"silvermont": []string{
 			"-march=slm",
 		},
+                "kabylake": []string{
+                },
 		"skylake": []string{
 		},
@@ -71,9 +70,11 @@ var (
 		"sse4_1": []string{"-msse4.1"},
 		"sse4_2": []string{"-msse4.2"},
 		"popcnt": []string{"-mpopcnt"},
-		"avx":    []string{"-mavx"},
-		"avx2":   []string{"-mavx2"},
-		"avx512": []string{"-mavx512"},
+                // Not all cases there is performance gain by enabling avx flags.
+                // Individual modules need to enable it
+		//"avx":    []string{"-mavx"},
+		//"avx2":   []string{"-mavx2"},
+		//"avx512": []string{"-mavx512"},
 		"aes_ni": []string{"-maes"},
 	}
 )
diff --git a/cc/config/x86_device.go b/cc/config/x86_device.go
index 31f0e9d..4c25241 100644
--- a/cc/config/x86_device.go
+++ b/cc/config/x86_device.go
@@ -71,14 +71,12 @@ var (
 			"-march=slm",
 			"-mfpmath=sse",
 		},
-                "kabylake": []string{
-			"-march=core-avx2",
-			"-mfpmath=sse",
-		},
-		"skylake": []string{
-			"-march=skylake",
-			"-mfpmath=sse",
-		},
+                "kabylake":[]string{
                          "-mfpmath=sse",
+                },
+                "skylake": []string{
+                        "-mfpmath=sse",
+               },
 		"stoneyridge": []string{
 			"-march=bdver4",
 			"-mfpmath=sse",
@@ -90,8 +88,10 @@ var (
 		"sse4":   []string{"-msse4"},
 		"sse4_1": []string{"-msse4.1"},
 		"sse4_2": []string{"-msse4.2"},
-		"avx":    []string{"-mavx"},
-		"avx2":   []string{"-mavx2"},
+                // Not all cases there is performance gain by enabling avx flags.
+                // Individual modules need to enable it
+		//"avx":    []string{"-mavx"},
+		//"avx2":   []string{"-mavx2"},
 		"aes_ni": []string{"-maes"},
 	}
 )
-- 
2.7.4

